cfme/configure/configuration/diagnostics_settings.py

```diff
@@ -0,0 +1,105 @@
+import attr
+
+from navmazing import NavigateToAttribute
+from widgetastic_patternfly import Button, Dropdown
+from widgetastic.widget import View, Table, RowNotFound
+
+from cfme.base.ui import ServerDiagnosticsView
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+
+# ============================ Diagnostic Server Workers ===========================
+
+class DiagnosticServerWorkersToolbar(View):
+    configuration = Dropdown('Configuration')
+    reload_button = Button(id='refresh_workers')
+
+
+class DiagnosticServerWorkersView(ServerDiagnosticsView):
+    toolbar = View.nested(DiagnosticServerWorkersToolbar)
+    workers_table = Table('//div[@id="diagnostics_workers"]//table')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.workers.is_displayed and
+            self.workers.is_active and
+            self.title.text == 'Diagnostics Server "{} [{}]" (current)'.format(
+                self.context['object'].name, self.context['object'].sid)
+        )
+
+
+@attr.s
+class DiagnosticWorker(BaseEntity):
+    """ A class representing Server DiagnosticWorker in the UI.
+
+         Args:
+             name : Worker name
+             description: Worker description
+    """
+    name = attr.ib()
+    description = attr.ib(default=None)
+
+    def get_all_worker_pids(self):
+        """ Returns a list of pids for worker """
+        view = navigate_to(self.parent, 'AllDiagnosticWorkers')
+        return (
+            {row.pid.text for row in view.workers_table.rows() if self.name in row.name.text}
+        )
+
+    def reload_worker(self, pid=None):
+        """ Reload workers
+
+            Returns: Workers pid(list)
+        """
+        if not pid:
+            pid = self.get_all_worker_pids()
+        view = self.create_view(DiagnosticServerWorkersView)
```

Comment:
Why only create_view and not navigate?

Reply:
We do navigation in get_all_worker_pids() method. But if you suggest navigation, I will update then

Reply:
aha, I see my mistake here, fixing this

Reply:
That only happens if pid is not passed, so in that case it could be on any screen. This at least needs a view.is_displayed before proceeding.